cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/AbstractReactiveLimitsOwnerConversion.java

```diff
@@ -32,49 +36,89 @@ protected void convertReactiveLimits(Generator g) {
+                    }
+
+                    Range<Double> qRange = fixedMinMaxQ("ReactiveCapabilityCurve Point", minQ, maxQ);
+
+                    // check if there is a previous point with same p
+                    Range<Double> prev = qRanges.get(p);
+                    if (prev != null) {
+                        if (prev.isConnected(qRange)) {
+                            fixed("reactive capability curve", String.format("point merged with another one with same p (%f)", p));
+                            qRanges.put(p, prev.span(qRange));
+                        } else {
+                            ignored("reactive capability curve point", String.format("another point with same p (%f) and a disconnected reactive range", p));
+                        }
+                    } else {
+                        qRanges.put(p, qRange);
+                    }
+                });
+
+                // create curve
+                ReactiveCapabilityCurveAdder rcca = g.newReactiveCapabilityCurve();
+                for (Map.Entry<Double, Range<Double>> e : qRanges.entrySet()) {
+                    double p = e.getKey();
+                    Range<Double> qRange = e.getValue();
+                    rcca.beginPoint()
+                            .setP(p)
+                            .setMinQ(qRange.lowerEndpoint())
+                            .setMaxQ(qRange.upperEndpoint())
+                            .endPoint();
+                }
+                rcca.add();
+            } else if (curveData.size() == 1) {
+                fixed("reactive capability curve", "only one point");
+                PropertyBag d = curveData.get(0);
                 double p = d.asDouble("xvalue");
                 double minQ = d.asDouble("y1value");
                 double maxQ = d.asDouble("y2value");
-                // Check the point is valid
-                if (Double.isNaN(p) || Double.isNaN(minQ) || Double.isNaN(maxQ)) {
-                    String reason = String.format("Incomplete point p, minQ, maxQ = %f, %f, %f", p, minQ, maxQ);
-                    ignored("ReactiveCapabilityCurvePoint", reason);
-                    return;
+                if (checkPointValidity(p, minQ, maxQ)) {
+                    Range<Double> qRange = fixedMinMaxQ("ReactiveCapabilityCurve Point", minQ, maxQ);
+                    g.newMinMaxReactiveLimits()
+                            .setMinQ(qRange.lowerEndpoint())
+                            .setMaxQ(qRange.upperEndpoint())
+                            .add();
                 }
-                double[] minMaxQ = fixedMinMaxQ("ReactiveCapabilityCurve Point", minQ, maxQ);
-                minQ = minMaxQ[0];
-                maxQ = minMaxQ[1];
-                rcca.beginPoint()
-                    .setP(p)
-                    .setMinQ(minQ)
-                    .setMaxQ(maxQ)
-                    .endPoint();
-            });
-            rcca.add();
+            } else {
+                missing("Empty curve");
```

Comment:
In this case, we should still try to read `minQ` and `maxQ` in the traditional way.

Reply:
Ok, done